tst/index.test.ts

```diff
@@ -293,15 +249,12 @@ describe("client", () => {
       "gets with ids",
       async ({ method, entityType, payload }) => {
         const mockIds = ["abc", "123"];
-        fetch.mockImplementation(() =>
-          Promise.resolve({
-            async json() {
-              return payload;
-            },
-          })
+        jest.spyOn(global, "fetch").mockImplementation(() =>
+          Promise.resolve(new Response(JSON.stringify(payload), { status: 200 }))
         );
 
         const client = new Client();
+        // @ts-expect-error idk, gang, its some deep typescript magic
```

Comment:
This is the end of my typescript knowledge. I'm not sure how to indicate to the compiler that all of the possible methods defined in the array above take an object with a key called `ids`. I can cut a ticket to address this if we're okay with merging as-is, or I can continue researching.

Reply:
Can't say my TypeScript knowledge is much better. I think we're okay to open a ticket for it and merge as is.

Reply:
I think the fix is in the function definitions themselves.

For example, one of the methods your test exercises is this:

`async getPlayers({ leagues = Object.values(LEAGUES), ids = [] } = {})`

If you roll over `ids` param, you'll notice TypeScript marked it as type `never[]` because it doesn't know what type of array that is -- which is also the error message you're seeing in your test.

When I went through those handful of functions and added `as string[]` (like example below) the error went away in your test.

`async getPlayers({ leagues = Object.values(LEAGUES), ids = [] as string[] } = {})`

You could also use `ids = [] as Array<string>` if you prefer, or add types for both parameters like so:

```
async getPlayers({
    leagues = Object.values(LEAGUES),
    ids = [],
  }: { leagues?: string[]; ids?: string[] } = {})
```

Reply:
@briangershon that totally resolved it! Thank you so much for your help!